lua/xbase/treesitter.lua

```diff
@@ -0,0 +1,153 @@
+local ts_utils = require("nvim-treesitter.ts_utils")
+local get_node_text = function(node)
+	return vim.treesitter.get_node_text(node, vim.api.nvim_get_current_buf())
+end
+local get_winr = vim.api.nvim_get_current_win
+local get_current_node = ts_utils.get_node_at_cursor
+local actions = {}
```

Comment:
Here get_winr and get_current_node are internal

Reply:
Somehow when I reviewed, I highlighted more lines. Was reffering to actions and how it's k value be a generator

Reply:
I see yea. I did that for add_modifier, do you think we should do that for all values in action? In which case it would look like:

```
actions = {
  extract_component = function()
    return extract_component()
  end
}
```

Reply:
Yes. That would be optimial and extendable when we need to pass options to `extract_component` in the future.

Reply:
🤔 hmm I'd avoid this, as it will introduce a lot of unwanted duplications.

You could change how user define actions, that will be pre-processesd before being passed to null_ls, e.g.

```lua
actions = {
    {
        title = "Add padding top 4",
        padding = {
           pos = "top",
           len = "4"
        }
    }
}
```  

Now, we are going into enabling users to decoratively describing what they want.  

What do you think? I feel more work 😀